Blocking crash on a Asus ZenFone 6 running Android 4.4.2

i plugged my phone and opened android monitor. here is the exception:

03-18 22:02:20.017 13381-13381/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.mapzen.erasermap, PID: 13381
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mapzen.erasermap/com.mapzen.erasermap.controller.MainActivity}: android.view.InflateException: Binary XML file line #14: Error inflating class com.mapzen.erasermap.view.CompassView
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2318)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396)
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:139)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1293)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:149)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5257)
                                                       at java.lang.reflect.Method.invokeNative(Native Method)
                                                       at java.lang.reflect.Method.invoke(Method.java:515)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609)
                                                       at dalvik.system.NativeStart.main(Native Method)
                                                    Caused by: android.view.InflateException: Binary XML file line #14: Error inflating class com.mapzen.erasermap.view.CompassView
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:620)
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:696)
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:492)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:353)
                                                       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:249)
                                                       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:106)
                                                       at com.mapzen.erasermap.controller.MainActivity.onCreate(MainActivity.kt:137)
                                                       at android.app.Activity.performCreate(Activity.java:5411)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2270)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396) 
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:139) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1293) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:149) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5257) 
                                                       at java.lang.reflect.Method.invokeNative(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:515) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609) 
                                                       at dalvik.system.NativeStart.main(Native Method) 
                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                       at java.lang.reflect.Constructor.constructNative(Native Method)
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:594)
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:696) 
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:492) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:353) 
                                                       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:249) 
                                                       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:106) 
                                                       at com.mapzen.erasermap.controller.MainActivity.onCreate(MainActivity.kt:137) 
                                                       at android.app.Activity.performCreate(Activity.java:5411) 
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087) 
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2270) 
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396) 
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:139) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1293) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:149) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5257) 
                                                       at java.lang.reflect.Method.invokeNative(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:515) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609) 
                                                       at dalvik.system.NativeStart.main(Native Method) 
                                                    Caused by: android.view.InflateException: Binary XML file line #4: Error inflating class <unknown>
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:620)
                                                       at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:56)
                                                       at android.view.LayoutInflater.onCreateView(LayoutInflater.java:669)
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:694)
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:462)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397)
                                                       at com.mapzen.erasermap.view.ButtonView.<init>(ButtonView.kt:23)
                                                       at com.mapzen.erasermap.view.CompassView.<init>(CompassView.kt:7)
                                                       at java.lang.reflect.Constructor.constructNative(Native Method) 
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:423) 
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:594) 
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:696) 
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:492) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:353) 
                                                       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:249) 
                                                       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:106) 
                                                       at com.mapzen.erasermap.controller.MainActivity.onCreate(MainActivity.kt:137) 
                                                       at android.app.Activity.performCreate(Activity.java:5411) 
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087) 
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2270) 
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396) 
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:139) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1293) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:149) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5257) 
                                                       at java.lang.reflect.Method.invokeNative(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:515) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609) 
                                                       at dalvik.system.NativeStart.main(Native Method) 
                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                       at java.lang.reflect.Constructor.constructNative(Native Method)
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:594)
                                                       at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:56) 
                                                       at android.view.LayoutInflater.onCreateView(LayoutInflater.java:669) 
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:694) 
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:462) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397) 
                                                       at com.mapzen.erasermap.view.ButtonView.<init>(ButtonView.kt:23) 
                                                       at com.mapzen.erasermap.view.CompassView.<init>(CompassView.kt:7) 
                                                       at java.lang.reflect.Constructor.constructNative(Native Method) 
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:423) 
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:594) 
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:696) 
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:492) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:353) 
                                                       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:249) 
                                                       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:106) 
                                                       at com.mapzen.erasermap.controller.MainActivity.onCreate(MainActivity.kt:137) 
                                                       at android.app.Activity.performCreate(Activity.java:5411) 
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087) 
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2270) 
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396) 
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:139) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1293) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:149) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5257) 
                                                       at java.lang.reflect.Method.invokeNative(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:515) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609) 
                                                       at dalvik.system.NativeStart.main(Native Method) 
                                                    Caused by: android.content.res.Resources$NotFoundException: File res/drawable/bg_white_gray_border.xml from drawable resource ID #0x7f02003e
                                                       at android.content.res.Resources.loadDrawable(Resources.java:2156)
                                                       at android.content.res.Resources.loadDrawable(Resources.java:2075)
                                                       at android.content.res.TypedArray.getDrawable(TypedArray.java:602)
                                                       at android.widget.ImageView.<init>(ImageView.java:129)
                                                       at android.widget.ImageView.<init>(ImageView.java:119)
                                                       at java.lang.reflect.Constructor.constructNative(Native Method) 
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:423) 
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:594) 
                                                       at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:56) 
                                                       at android.view.LayoutInflater.onCreateView(LayoutInflater.java:669) 
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:694) 
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:462) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397) 
                                                       at com.mapzen.erasermap.view.ButtonView.<init>(ButtonView.kt:23) 
                                                       at com.mapzen.erasermap.view.CompassView.<init>(CompassView.kt:7) 
                                                       at java.lang.reflect.Constructor.constructNative(Native Method) 
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:423) 
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:594) 
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:696) 
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:755) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:492) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:397) 
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:353) 
                                                       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:249) 
                                                       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:106) 
                                                       at com.mapzen.erasermap.controller.MainActivity.onCreate(MainActivity.kt:137) 
                                                       at android.app.Activity.performCreate(Activity.java:5411) 
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087) 
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2270) 
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2396) 
                                                       at android.app.ActivityThread.access$800(ActivityThread.java:139) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1293) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:149) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5257) 
                                                       at java.lang.reflect.Method.invokeNative(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:515) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:609) 
                                                       at dalvik.system.NativeStart.main(Native Method) 
                                                    Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #16: <item> tag requires a 'drawable' attribute or child tag defining a drawable
                                                   	at android.graphics.drawable.L

--- a/app/src/main/res/drawable/bg_black_gray_border.xml
+++ b/app/src/main/res/drawable/bg_black_gray_border.xml
@@ -12,5 +12,5 @@
                 android:color="@color/gray" />
         </shape>
     </item>
-    <item android:drawable="?android:selectableItemBackground"/>
+    <item android:drawable="@drawable/bg_ripple"/>
 </layer-list>

--- a/app/src/main/res/drawable/bg_blue_gray_border.xml
+++ b/app/src/main/res/drawable/bg_blue_gray_border.xml
@@ -13,5 +13,5 @@
 
         </shape>
     </item>
-    <item android:drawable="?android:selectableItemBackground"/>
+    <item android:drawable="@drawable/bg_ripple"/>
 </layer-list>

--- a/app/src/main/res/drawable/bg_ripple.xml
+++ b/app/src/main/res/drawable/bg_ripple.xml
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@color/gray" android:state_focused="true"/>
+    <item android:drawable="@color/gray" android:state_pressed="true"/>
+    <item android:drawable="@android:color/transparent"/>
+</selector>

--- a/app/src/main/res/drawable/bg_white_gray_border.xml
+++ b/app/src/main/res/drawable/bg_white_gray_border.xml
@@ -13,5 +13,5 @@
 
         </shape>
     </item>
-    <item android:drawable="?android:selectableItemBackground"/>
+    <item android:drawable="@drawable/bg_ripple"/>
 </layer-list>

